core/src/com/unciv/ui/worldscreen/Minimap.kt

```diff
@@ -120,13 +123,65 @@ class Minimap(val mapHolder: WorldMapHolder) : Table(){
 class MinimapHolder(mapHolder: WorldMapHolder): Table() {
     val minimap = Minimap(mapHolder)
     val worldScreen = mapHolder.worldScreen
+    private data class ToggleButtonInfo(val actor: Actor, val getIt: ()->Boolean, val setIt: (Boolean)->Unit)
+    private val toggleButtonInfo: EnumMap<MinimapToggleButtons,ToggleButtonInfo> = EnumMap<MinimapToggleButtons,ToggleButtonInfo>(MinimapToggleButtons::class.java)
 
     init {
         add(getToggleIcons()).align(Align.bottom)
         add(getWrappedMinimap())
         pack()
     }
 
+    enum class MinimapToggleButtons {
+        YIELD {
+            override val icon = "Food"
+        },
+        WORKED {
+            override val icon = "Population"
+        },
+        RESOURCES {
+            override val icon = "ResourceIcons/Cattle"
+        };
+        abstract val icon: String
+    }
+
+    // "Api" when external code wants to toggle something together with our buttons
+    fun getToggleButton(which: MinimapToggleButtons): Boolean {
```

Comment:
These functions need renaming stat. 'which' as a function parameter? 'info'? 'getIt'? None of this helps you understand what the underlying data is or how it's being used. Function is named 'getToggleButton' and yet returns something completely different.

Sorry for being blunt, but this looks like an anti-example I would give in a SOLID lecture to new programmers :/